futaba/cogs/info/core.py

```diff
@@ -0,0 +1,190 @@
+#
+# cogs/info/core.py
+#
+# futaba - A Discord Mod bot for the Programming server
+# Copyright (c) 2017 Jake Richardson, Ammon Smith, jackylam5
+#
+# futaba is available free of charge under the terms of the MIT
+# License. You are free to redistribute and/or modify it under those
+# terms. It is distributed in the hopes that it will be useful, but
+# WITHOUT ANY WARRANTY. See the LICENSE file for more details.
+#
+
+'''
+Informational commands that make finding and gathering data easier.
+'''
+
+import asyncio
+import logging
+from datetime import datetime
+from itertools import islice
+
+import discord
+from discord.ext import commands
+
+from futaba.enums import Reactions
+from futaba.parse import get_user_id, similar_user_ids
+from futaba.utils import fancy_timedelta
+
+logger = logging.getLogger(__package__)
+
+__all__ = [
+    'Info',
+]
+
+class Info:
+    '''
+    Cog for informational commands.
+    '''
+
+    def __init__(self, bot):
+        self.bot = bot
+
+    @commands.command(name='uinfo', aliases=['userinfo'])
+    async def uinfo(self, ctx, *, name: str = None):
+        '''
+        Fetch information about a user, whether they are in the guild or not.
+        If no argument is passed, the caller is checked instead.
+        '''
+
+        if name is None:
+            name = str(ctx.message.author.id)
+
+        logger.info("Running uinfo on '%s'", name)
+
+        id = get_user_id(name, self.bot.users)
+        if id is None:
+            logger.debug("No user ID found!")
+            await Reactions.FAIL.add(ctx.message)
+            return
+
+        logger.debug("Fetched user ID is %d", id)
+
+        user = None
+        if ctx.guild is not None:
+            user = ctx.guild.get_member(id)
+
+        if user is None:
+            user = self.bot.get_user(id)
+            if user is None:
+                user = await self.bot.get_user_info(id)
+                if user is None:
+                    logger.debug("No user with that ID found!")
+                    await Reactions.FAIL.add(ctx.message)
+                    return
+
+        logger.debug("Found user! %r", user)
+
+        # Status
+        if getattr(user, 'status', None):
+            status = 'do not disturb' if user.status == discord.Status.dnd else user.status
```

Comment:
Can we make the status capitalised as it would look nicer in my opinion. Can we also do the same to voice states?